components/navigation/NavBar.tsx

```diff
@@ -223,7 +223,7 @@ export default function NavBar({ className = '', hideLogo = false }: NavBarProps
               className='mr-2 flex items-center space-x-2 rounded-md p-2 text-left text-gray-400 transition duration-150 ease-in-out hover:bg-gray-100 hover:text-gray-500 focus:bg-gray-100 focus:text-gray-500 focus:outline-none'
               aria-label='Open Search'
             >
-              <IconLoupe />
+              {() => <IconLoupe />}
```

Comment:
@amanbhoria can you please elaborate this change a little

Reply:
@sambhavgupta0705 As you can see in the useEffect, type of children is expected as a function, but it was a node earlier. That's the reason why Algolia search icon was not visible. 

![image](https://github.com/user-attachments/assets/b24832dd-381b-4012-8036-9e0ca9e351aa)

Reply:
@amanbhoria Can we change the useEffect function then, to fix this rendering?

Reply:
![image](https://github.com/user-attachments/assets/d4f45a0c-dc70-4824-af04-fa6473224c03)

@akshatnema As it can handle both the searches on landing page simultaneously. Can we do something like that?

Reply:
Yeah, can you please try that and see if we are facing any errors or not.

Reply:
As far as the search's concern, everything seems to be working.